modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/IgniteTableImpl.java

```diff
@@ -45,6 +47,25 @@ public IgniteTableImpl(String name, int id, int version, TableDescriptor desc,
         this.partitions = partitions;
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public Supplier<PartitionCalculator> partitionCalculator() {
+        return () -> {
+            int fieldCnt = descriptor().distribution().getKeys().size();
+            NativeType[] fieldTypes = new NativeType[fieldCnt];
+
+            int[] colocationColumns = descriptor().distribution().getKeys().toIntArray();
+
+            for (int i = 0; i < fieldCnt; i++) {
+                ColumnDescriptor colDesc = descriptor().columnDescriptor(colocationColumns[i]);
+
+                fieldTypes[i] = colDesc.physicalType();
+            }
```

Comment:
Is it possible to cache `fieldTypes` array in the IgniteTableImpl instance and reuse it?